solr/core/src/java/org/apache/solr/handler/component/CloudReplicaSource.java

```diff
@@ -170,10 +170,9 @@ private List<String> findReplicas(
               .filter(replica -> replica.isActive(clusterState.getLiveNodes()))
               .filter(
                   replica ->
-                      !builder.onlyNrt
-                          || (replica.getType() == Replica.Type.NRT
-                              || (replica.getType() == Replica.Type.TLOG
-                                  && isShardLeader.test(replica))))
+                      !builder.onlyRealTime
+                          || (replica.getType().follower && !replica.getType().followerSkipCommit)
+                          || (replica.getType().leaderEligible && isShardLeader.test(replica)))
```

Comment:
Equivalent to `replica.getType() == Replica.Type.TLOG && isShardLeader.test(replica)`

Reply:
if the replica is shard leader then obviously it is also leader eligible, right?
Before the change checking that the type is TLOG was already useless since PULL can't be leader.

Reply:
The goal is to avoid calling the costly `isShardLeader` predicate if we know the replica type cannot be a leader.